internal/pass/pairing/proxy.go

```diff
@@ -0,0 +1,76 @@
+package pairing
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/twofas/2fas-server/internal/common/logging"
+)
+
+type Proxy struct {
+	wsStore wsStore
+}
+
+func NewProxy() *Proxy {
+	return &Proxy{wsStore: NewWSMemoryStore()}
+}
+
+type wsStore interface {
+	SetMobileConn(ctx context.Context, deviceID string, conn *websocket.Conn)
+	GetMobileConn(ctx context.Context, deviceID string) (*websocket.Conn, bool)
+}
+
+func (p *Proxy) ServeExtensionProxyToMobileWS(w http.ResponseWriter, r *http.Request, extID, deviceID string) {
```

Comment:
I think we should do a better work at supporting retries from both mobile and browser. Let's have an offline discussion about what we exactly want to achieve in this area.